description.md

```diff
@@ -38,6 +38,73 @@ Put differently, it shows how each content type is to be used in the various tea
 
 ### Assignments
 
+Assignments are unassisted practical activities that learners undertake individually or in teams over a longer timespan.
+The primary objective of an assignment is to create a compelling scenario that allows learners to apply the concepts they have learned.
+
+#### Storing Assignments
+
+The central component of an assignment is the assignment statement, serving the purpose of offering a detailed explanation for each task.
+Additionally, it may offer an introduction, details about the evaluation process, and further referrences.
+Typically, this content is stored in either markdown format or as a PDF, allowing for the incorporation of media files.
+
+An assignment might contain any of the following:
+
+- Utility Files
+
+  These files can be exemplars or code snippets meant to facilitate solution development.
+  They often come into play when addressing topics tangential to the main assignment task.
+  These files are designed for utility purposes and should remain unaltered by learners; they are not subject to evaluation.
+
+- Skeleton Files
+
+  These serve as the starting point of the assignment and establish a similar pattern, which is crucial for automated testing.
+  Depending on the subject matter, skeleton files range from templates and fill-in-the-blank outlines to simple guidelines.
+
+- Automated Evaluation Program
```

Comment:
I would discuss a bit the possibility of such evaluation programs and highlight the fact that this is not applicable for open ended assignments such as essays, proofs etc.